autoload/cmake/buildsys.vim

```diff
@@ -375,6 +358,7 @@ function! s:SetCurrentConfig(config) abort
     " Save project's current config and build dir.
     let l:state = {'config': a:config, 'build_dir': l:path}
     call s:state.WriteProjectState(s:buildsys.project_root, l:state)
+    call s:RefreshTargets()
```

Comment:
Because switching config may change available targets depending on CMake configuration (e.g. a test target that only shows up in the Debug config).

Reply:
True, but we don't always need to refresh targets in this function. For instance, the first time we call `s:SetCurrentConfig()` is inside `s:ProcessBuildConfig()`, when we're processing the command-line arguments passed by the user with `:CMakeGenerate`. At that point, the build system does not exist yet, so conceptually there's not point in trying to refresh targets.

I would suggest to explicitly call `s:RefreshTargets()` where really needed (`s:buildsys.Init()` and `s:buildsys.Switch()`). This would also make the logic more clear for someone who reads the code in the future, in my opinion.

Reply:
Fair enough